README.md

```diff
@@ -1,48 +1,207 @@
+
+# Lifecycle Timing
+
+Before proceeding, keep the answer to this question in mind:
+
+**What is the difference between `#new()` and `#create()`?**
+
+If you've forgotten, `#new()` instantiates a new ActiveRecord model *without*
+saving it to the database, whereas `#create()` immediately attempts to save it,
+as if you had called `#new()` and then `#save()`.
+
+**Database activity triggers validation**. An ActiveRecord model instantiated
+with `#new()` will *not* be validated, because no attempt to write to the
+database was made.
+
+The only way to trigger validation without touching the database is to call the
+`#valid?()` method.
+
+For a full list of methods that trigger validation, see [Section
+4][ar_callbacks_4] of the Rails Guide for Active Record Callbacks. Don't worry
+about the rest of the information in that guide just yet; we'll go into
+callbacks later!
+
+[ar_callbacks_4]: http://guides.rubyonrails.org/active_record_callbacks.html#running-callbacks
+
+# Validation Failure
+
+Here it is, the moment of truth. What can we do when a record fails validation?
+
+## How can you tell when a record fails validation?
+
+**Pay attention to return values!**
+
+By default, ActiveRecord does not raise an exception when validation fails.
+The database operation method will simply return `false` and decline to update
+the database.
+
+Every database method has a sister method with a `!` at the end which will raise
+an exception (`#create!()` instead of `#create()` and so on).
+
+And of course, you can always check manually with `#valid?()`.
+
+## Finding out why validations failed
+
+To find out what went wrong, you can look at the model's `#errors()` object.
+
+You can check all errors at once by examining `errors.messages`.
+
+You can check one attribute at a time by passing the name to `errors` as a key,
+like so:
+
+```ruby
+person.errors[:name]
+```
+
+
+# Displaying Validation Errors in Views
+
+See [Section 8][ar_validations_8] of the Rails Guide for an example of how to
+use the `ActiveModel::Errors#full_messages()` helper, reproduced here for
+convenience:
+
+[ar_validations_8]: http://guides.rubyonrails.org/active_record_validations.html#displaying-validation-errors-in-views
+
+```erb
+<% if @article.errors.any? %>
+  <div id="error_explanation">
+    <h2>
+      <%= pluralize(@article.errors.count, "error") %>
+      prohibited this article from being saved:
+    </h2>
+
+    <ul>
+    <% @article.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
+```
 
-Introduce the concept of validation, our models protect our DB and can inforce requirements for the attributes.
+This constructs more complete-looking sentences from the more terse messages
+available in `errors.messages`.
 
-The validates macro.
 
-Validates presence :name
+# Custom Validators
 
-When do validations run?
+There are three ways to implement custom validators, with examples in [Section
```

Comment:
I would actually present from simplest to most complex. That's how we want them to approach coding. Also explaining that the student will mostly use the custom instance method version and giving an example of how that works.